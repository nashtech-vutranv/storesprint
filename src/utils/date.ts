import moment from 'moment'
import {FORMAT_DATE, FORMATE_DATE_NO_HOURS} from '../constants'

export const formatDate = (date: string, format: string = FORMAT_DATE) => {
  return moment(date).format(format)
}

export const formatDateNoHours = (date: string, format: string = FORMATE_DATE_NO_HOURS) => {
  return moment(date).format(format)
}
