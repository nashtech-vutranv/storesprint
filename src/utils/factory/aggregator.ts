import faker from '@faker-js/faker'

export const mockAggregator = () => {
  return {
    id: faker.random.word(),
    name: faker.random.word(),
    version: faker.datatype.number(),
    status: faker.datatype.string(),
  }
}

export const mockAggregators = () => {
  return {
    data: {
      content: Array.apply(null, Array(10)).map(() => mockAggregator()),
      totalElements: 10,
    },
  }
}
