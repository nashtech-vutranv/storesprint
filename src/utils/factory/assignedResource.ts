import faker from '@faker-js/faker'

export const mockAssignedResource = () => {
  return {
    data: {
      createdAt: faker.datatype.string(),
      id: faker.datatype.string(),
      modifiedAt: faker.datatype.string(),
      organizationResponse: {
        additionalInformation: faker.datatype.string(),
        addressLine1: faker.datatype.string(),
        addressLine2: null,
        addressLine3: null,
        city: faker.datatype.string(),
        contactEmailAddress: faker.datatype.string(),
        contactName: faker.datatype.string(),
        contactPhoneNumber: faker.datatype.string(),
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        postCode: faker.datatype.string(),
        registrationNumber: faker.datatype.string(),
        status: faker.datatype.string(),
        vatRegistrationNumber: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      siteResponse: {
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        organizationId: faker.datatype.string(),
        status: faker.datatype.string(),
        url: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      userResponse: null,
      version: faker.datatype.number(),
    },
  }
}
