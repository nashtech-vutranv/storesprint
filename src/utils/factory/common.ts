import {IPaginationResponse} from '../../interface'

export const mockOption = () => ({
  value: '',
  label: '',
})

export const mockPaginationResponse = (): IPaginationResponse => {
  return {
    pageable: {
      sort: {
        empty: true,
        sorted: false,
        unsorted: true,
      },
      offset: 0,
      pageSize: 3,
      pageNumber: 0,
      unpaged: false,
      paged: true,
    },
    last: false,
    totalPages: 2,
    totalElements: 5,
    number: 0,
    size: 3,
    first: true,
    numberOfElements: 3,
    sort: {
      empty: true,
      sorted: false,
      unsorted: true,
    },
    empty: false,
  }
}
