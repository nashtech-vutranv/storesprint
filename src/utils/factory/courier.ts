import faker from '@faker-js/faker'
import {ICourierMapping} from '../../interface'

export const mockCourier = (): ICourierMapping => {
  return {
    id: faker.datatype.string(),
    createdAt: faker.datatype.string(),
    modifiedAt: faker.datatype.string(),
    mmsValue: faker.datatype.string(),
    version: faker.datatype.number(),
    mappingTarget: {
        id: faker.datatype.string(),
        mappingTypeId: faker.datatype.string(),
        marketplaceCode: faker.datatype.string(),
        marketplaceTypeId: faker.datatype.string(),
        marketplaceValue: faker.datatype.string(),
    },
    mappingType: {
        id: faker.datatype.string(),
        name: faker.datatype.string(),
    },
    mappingTypeId: faker.datatype.string(),
    marketplaceType: {
        id: faker.datatype.string(),
        name: faker.datatype.string(),
    },
    marketplaceTypeId: faker.datatype.string(),
    marketplaceValue: faker.datatype.string(),
  }
}

export const mockCouriers = (): ICourierMapping[] => {
    return Array.apply(null, Array(10)).map(() =>mockCourier()) 
}
