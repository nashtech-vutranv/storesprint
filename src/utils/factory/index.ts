export * from './organization'
export * from './warehouse'
export * from './aggregator'
export * from './marketplaces'
export * from './currency'
export * from './relationshipAM'
export * from './marketplaceRelationships'
export * from './product'
export * from './order'
export * from './listingStatus'
export * from './common'
export * from './defaultPropertyValue'
export * from './mappings'
export * from './productProperties'
export * from './assignedResource'
export * from './productCategoryMappings'
export * from './productPropertyMappings'
export * from './courier'
export * from './valueMapping'
export * from './courierMappings'
export * from './returnsrefunds'
export * from './deliveryTypeMappings'
