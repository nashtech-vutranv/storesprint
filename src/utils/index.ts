export * from './array'
export * from './layout'
export * from './date'
export * from './conditionPromise'
