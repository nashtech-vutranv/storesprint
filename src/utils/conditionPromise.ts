import {mockOrganizations, mockSites, mockSiteLocales} from './factory/organization'
import {mockMarketplaces, mockCalculateAssignMarketplaces} from './factory/marketplaces'
import {mockAssignedProducts, mockAssignedProductsResponse} from './factory/product'
import {mockWarehouses} from './factory/warehouse'
import {mockMarketplaceRelationship} from './factory/marketplaceRelationships'

type IMockType = 'organization' | 'site' | 'siteLocale' |
 'marketplace' | 'calculateMarketplace' | 'warehouse' 
 | 'marketplaceRelationship' | 'assignedProduct' | 'assignedProductResponse'

export const conditionPromise = (objectFilter: unknown, typeMock: IMockType, objectPaginationQuery?: any) => {
  if (objectFilter || (objectFilter && objectPaginationQuery)) {
    switch (typeMock) {
      case 'organization': return Promise.resolve(mockOrganizations())
      case 'site': return Promise.resolve(mockSites())
      case 'siteLocale': return Promise.resolve(mockSiteLocales())
      case 'marketplace': return Promise.resolve(mockMarketplaces())
      case 'calculateMarketplace': return Promise.resolve(mockCalculateAssignMarketplaces())
      case 'warehouse': return mockWarehouses()
      case 'marketplaceRelationship': return mockMarketplaceRelationship()
      case 'assignedProduct': return Promise.resolve(mockAssignedProducts())
      case 'assignedProductResponse': return Promise.resolve(mockAssignedProductsResponse())

      default: return {}
    }
  } else return Promise.reject()
}