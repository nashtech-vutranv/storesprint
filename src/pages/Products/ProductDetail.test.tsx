import {render, waitFor} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import '@testing-library/jest-dom'
import {mockProductDetail, mockPropertyGeneric, mockOption} from '../../utils/factory'
import ProductDetail from './ProductDetail'

beforeEach(() => {
  window.history.pushState({
    selectedOrganizations: [mockOption()],
    selectedSites: [mockOption()],
    selectedLocales: [mockOption()],
    currentStatus: null,
    eventKey: '0'
  }, '', '')})

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({productId: 'test_id'}),
}))

test('should match snapshot with successfully data from api', async () => {
  jest.mock('../../services/ProductService', () => {
    const instance = jest.fn().mockReturnValue({
      getProductById: jest
        .fn()
        .mockImplementation(() => Promise.resolve(mockProductDetail())),
    })
    return instance
  })

  jest.mock('../../services/ProductService', () => {
    const instance = jest.fn().mockReturnValue({
      getGenericProperties: jest
        .fn()
        .mockImplementation(() => Promise.resolve(mockPropertyGeneric().property)),
    })
    return instance
  })

  const tree = render(
    <MemoryRouter>
      <ProductDetail />
    </MemoryRouter>
  )


  await waitFor(() => {
    expect(tree).toMatchSnapshot()
    expect(
      tree.queryByText(/product_detail_organizationErpId/i)
    ).toBeInTheDocument()
    expect(
      tree.queryByText(/product_detail_organizationName/i)
    ).toBeInTheDocument()
    expect(tree.queryByText(/product_detail_productId/i)).toBeInTheDocument()
    expect(tree.queryByText(/product_detail_productName/i)).toBeInTheDocument()
    expect(tree.queryByText(/product_detail_productErpId/i)).toBeInTheDocument()
    expect(
      tree.queryByText(/product_detail_tab_stockLevel/i)
    ).toBeInTheDocument()
    expect(tree.queryByText(/product_detail_tab_price/i)).toBeInTheDocument()
  })
  
  const organizationErpIdInput = tree.container.querySelector(
    'input[name="organizationErpId"]'
  ) as Element
  await waitFor(() => {
    expect(organizationErpIdInput).toHaveAttribute('disabled')
  })
  
  const organizationNameInput = tree.container.querySelector(
    'input[name="organizationName"]'
  ) as Element
  await waitFor(() => {
    expect(organizationNameInput).toHaveAttribute('disabled')
  })
  
  const idInput = tree.container.querySelector('input[name="id"]') as Element
  await waitFor(() => {
    expect(idInput).toHaveAttribute('disabled')
  })
  
  const erpIdInput = tree.container.querySelector(
    'input[name="erpId"]'
  ) as Element
  await waitFor(() => {
    expect(erpIdInput).toHaveAttribute('disabled')
  })
})
