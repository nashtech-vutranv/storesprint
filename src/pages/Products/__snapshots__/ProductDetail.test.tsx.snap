// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with successfully data from api 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-form mt-3 card"
      >
        <div
          class="card-header"
        >
          <h4
            class="card-form__title"
          >
            product_detail_title
          </h4>
        </div>
        <div
          class="card-body"
        >
          <form
            autocomplete="on"
            class="form-layout"
          >
            <div
              class="row"
            >
              <div
                class="container-fluid"
              >
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      for="organizationErpId"
                    >
                      product_detail_organizationErpId
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      aria-label="input-text"
                      class="p-inputtext p-component p-disabled w-full p-1"
                      disabled=""
                      id="organizationErpId"
                      name="organizationErpId"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="container-fluid"
              >
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      for="organizationName"
                    >
                      product_detail_organizationName
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      aria-label="input-text"
                      class="p-inputtext p-component p-disabled w-full p-1"
                      disabled=""
                      id="organizationName"
                      name="organizationName"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="container-fluid"
              >
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      for="id"
                    >
                      product_detail_productId
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      aria-label="input-text"
                      class="p-inputtext p-component p-disabled w-full p-1"
                      disabled=""
                      id="id"
                      name="id"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="container-fluid"
              >
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      for="erpId"
                    >
                      product_detail_productErpId
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      aria-label="input-text"
                      class="p-inputtext p-component p-disabled w-full p-1"
                      disabled=""
                      id="erpId"
                      name="erpId"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="container-fluid"
              >
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class="required"
                      for="name"
                    >
                      product_detail_productName
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      aria-label="input-text"
                      class="p-inputtext p-component w-full p-1"
                      id="name"
                      maxlength="255"
                      name="name"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="mt-4 row"
            >
              <div
                class="container-fluid"
              >
                <div
                  class="bg-nav-pills nav nav-pills nav-justified"
                  role="tablist"
                >
                  <div
                    class="nav-item"
                  >
                    <a
                      aria-controls="react-aria2064558450-1-tabpane-product_detail_tab_generic_properties"
                      aria-selected="true"
                      class="nav-link active"
                      data-rr-ui-event-key="product_detail_tab_generic_properties"
                      href="/"
                      id="react-aria2064558450-1-tab-product_detail_tab_generic_properties"
                      role="tab"
                    >
                      <i
                        class="d-md-none d-block me-1"
                      />
                      <span
                        class="d-none d-md-block"
                      >
                        product_detail_tab_generic_properties
                      </span>
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      aria-selected="false"
                      class="nav-link"
                      data-rr-ui-event-key="product_detail_tab_locale_properties"
                      href="/"
                      id="react-aria2064558450-1-tab-product_detail_tab_locale_properties"
                      role="tab"
                      tabindex="-1"
                    >
                      <i
                        class="d-md-none d-block me-1"
                      />
                      <span
                        class="d-none d-md-block"
                      >
                        product_detail_tab_locale_properties
                      </span>
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      aria-selected="false"
                      class="nav-link"
                      data-rr-ui-event-key="product_detail_tab_stockLevel"
                      href="/"
                      id="react-aria2064558450-1-tab-product_detail_tab_stockLevel"
                      role="tab"
                      tabindex="-1"
                    >
                      <i
                        class="d-md-none d-block me-1"
                      />
                      <span
                        class="d-none d-md-block"
                      >
                        product_detail_tab_stockLevel
                      </span>
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      aria-selected="false"
                      class="nav-link"
                      data-rr-ui-event-key="product_detail_tab_price"
                      href="/"
                      id="react-aria2064558450-1-tab-product_detail_tab_price"
                      role="tab"
                      tabindex="-1"
                    >
                      <i
                        class="d-md-none d-block me-1"
                      />
                      <span
                        class="d-none d-md-block"
                      >
                        product_detail_tab_price
                      </span>
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      aria-selected="false"
                      class="nav-link"
                      data-rr-ui-event-key="product_detail_tab_marketplace"
                      href="/"
                      id="react-aria2064558450-1-tab-product_detail_tab_marketplace"
                      role="tab"
                      tabindex="-1"
                    >
                      <i
                        class="d-md-none d-block me-1"
                      />
                      <span
                        class="d-none d-md-block"
                      >
                        product_detail_tab_marketplace
                      </span>
                    </a>
                  </div>
                </div>
                <div
                  class="tab-content"
                >
                  <div
                    aria-labelledby="react-aria2064558450-1-tab-product_detail_tab_generic_properties"
                    class="fade tab-pane active show"
                    id="react-aria2064558450-1-tabpane-product_detail_tab_generic_properties"
                    role="tabpanel"
                  >
                    <br />
                    <div
                      class="d-flex justify-content-end"
                    >
                      <button
                        class="btn btn-info"
                        type="button"
                      >
                        <i
                          class="pi pi-plus"
                        />
                         
                        <span>
                          product_detail_tab_generic_properties_add_label
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="mt-4 row"
            >
              <div
                class="col"
              >
                <div
                  class="d-flex justify-content-between"
                >
                  <button
                    class="me-2 btn btn-danger"
                    type="button"
                  >
                    common_confirm_cancel
                  </button>
                  <button
                    class="me-2 btn btn-success"
                    disabled=""
                    type="button"
                  >
                    common_confirm_save
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-form mt-3 card"
    >
      <div
        class="card-header"
      >
        <h4
          class="card-form__title"
        >
          product_detail_title
        </h4>
      </div>
      <div
        class="card-body"
      >
        <form
          autocomplete="on"
          class="form-layout"
        >
          <div
            class="row"
          >
            <div
              class="container-fluid"
            >
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    for="organizationErpId"
                  >
                    product_detail_organizationErpId
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    aria-label="input-text"
                    class="p-inputtext p-component p-disabled w-full p-1"
                    disabled=""
                    id="organizationErpId"
                    name="organizationErpId"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="container-fluid"
            >
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    for="organizationName"
                  >
                    product_detail_organizationName
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    aria-label="input-text"
                    class="p-inputtext p-component p-disabled w-full p-1"
                    disabled=""
                    id="organizationName"
                    name="organizationName"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="container-fluid"
            >
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    for="id"
                  >
                    product_detail_productId
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    aria-label="input-text"
                    class="p-inputtext p-component p-disabled w-full p-1"
                    disabled=""
                    id="id"
                    name="id"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="container-fluid"
            >
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    for="erpId"
                  >
                    product_detail_productErpId
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    aria-label="input-text"
                    class="p-inputtext p-component p-disabled w-full p-1"
                    disabled=""
                    id="erpId"
                    name="erpId"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="container-fluid"
            >
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class="required"
                    for="name"
                  >
                    product_detail_productName
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    aria-label="input-text"
                    class="p-inputtext p-component w-full p-1"
                    id="name"
                    maxlength="255"
                    name="name"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="mt-4 row"
          >
            <div
              class="container-fluid"
            >
              <div
                class="bg-nav-pills nav nav-pills nav-justified"
                role="tablist"
              >
                <div
                  class="nav-item"
                >
                  <a
                    aria-controls="react-aria2064558450-1-tabpane-product_detail_tab_generic_properties"
                    aria-selected="true"
                    class="nav-link active"
                    data-rr-ui-event-key="product_detail_tab_generic_properties"
                    href="/"
                    id="react-aria2064558450-1-tab-product_detail_tab_generic_properties"
                    role="tab"
                  >
                    <i
                      class="d-md-none d-block me-1"
                    />
                    <span
                      class="d-none d-md-block"
                    >
                      product_detail_tab_generic_properties
                    </span>
                  </a>
                </div>
                <div
                  class="nav-item"
                >
                  <a
                    aria-selected="false"
                    class="nav-link"
                    data-rr-ui-event-key="product_detail_tab_locale_properties"
                    href="/"
                    id="react-aria2064558450-1-tab-product_detail_tab_locale_properties"
                    role="tab"
                    tabindex="-1"
                  >
                    <i
                      class="d-md-none d-block me-1"
                    />
                    <span
                      class="d-none d-md-block"
                    >
                      product_detail_tab_locale_properties
                    </span>
                  </a>
                </div>
                <div
                  class="nav-item"
                >
                  <a
                    aria-selected="false"
                    class="nav-link"
                    data-rr-ui-event-key="product_detail_tab_stockLevel"
                    href="/"
                    id="react-aria2064558450-1-tab-product_detail_tab_stockLevel"
                    role="tab"
                    tabindex="-1"
                  >
                    <i
                      class="d-md-none d-block me-1"
                    />
                    <span
                      class="d-none d-md-block"
                    >
                      product_detail_tab_stockLevel
                    </span>
                  </a>
                </div>
                <div
                  class="nav-item"
                >
                  <a
                    aria-selected="false"
                    class="nav-link"
                    data-rr-ui-event-key="product_detail_tab_price"
                    href="/"
                    id="react-aria2064558450-1-tab-product_detail_tab_price"
                    role="tab"
                    tabindex="-1"
                  >
                    <i
                      class="d-md-none d-block me-1"
                    />
                    <span
                      class="d-none d-md-block"
                    >
                      product_detail_tab_price
                    </span>
                  </a>
                </div>
                <div
                  class="nav-item"
                >
                  <a
                    aria-selected="false"
                    class="nav-link"
                    data-rr-ui-event-key="product_detail_tab_marketplace"
                    href="/"
                    id="react-aria2064558450-1-tab-product_detail_tab_marketplace"
                    role="tab"
                    tabindex="-1"
                  >
                    <i
                      class="d-md-none d-block me-1"
                    />
                    <span
                      class="d-none d-md-block"
                    >
                      product_detail_tab_marketplace
                    </span>
                  </a>
                </div>
              </div>
              <div
                class="tab-content"
              >
                <div
                  aria-labelledby="react-aria2064558450-1-tab-product_detail_tab_generic_properties"
                  class="fade tab-pane active show"
                  id="react-aria2064558450-1-tabpane-product_detail_tab_generic_properties"
                  role="tabpanel"
                >
                  <br />
                  <div
                    class="d-flex justify-content-end"
                  >
                    <button
                      class="btn btn-info"
                      type="button"
                    >
                      <i
                        class="pi pi-plus"
                      />
                       
                      <span>
                        product_detail_tab_generic_properties_add_label
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="mt-4 row"
          >
            <div
              class="col"
            >
              <div
                class="d-flex justify-content-between"
              >
                <button
                  class="me-2 btn btn-danger"
                  type="button"
                >
                  common_confirm_cancel
                </button>
                <button
                  class="me-2 btn btn-success"
                  disabled=""
                  type="button"
                >
                  common_confirm_save
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
